### ISSUE TITLE
Network architecture types

### ISSUE THREAD

**Issue by ShilinC** [kashyap7x/QuadNet#3]
Hi,

I found several different architectures in the code including QuadNet and QGN. I guess the QuadNet refers to graph CNN and QGN is the proposed arch in the paper? Also, I found that in QGN there are multiple architecture options. I guess "QGN_dense_resnet" and "resnetXXX_transpose" means standard dense 2D convolution? "QGN_resnet" and "resnetXXX_transpose_sparse" means Submanifold Sparse convolution? If I understand correctly, it's better to use the sparse version since the sparse convnet is much faster than its dense counterpart?

Thanks!!!

**Comment by kashyap7x** [comment]
Your understanding of the architectures is correct. However, on the hardware we used at the time of submission, the sparse and dense convnets had similar execution time, and the sparse model is even a bit slower for low resolution inputs. This could be alleviated by designing a custom sparse convolution kernel specifically for quadtrees, though we did not explore this direction. The SparseConvNet library has more recently been updated with some fixes/optimizations which may help.

The main advantage of the sparse models is the reduction in memory consumption, which allows training with higher input resolution or batch size than dense models.
